### PR TITLE
Agreed changes

### PR DESCRIPTION
**Pull request by dudycooly**
1-  OrphanResource call method 
2- and some refactoring

### CODE REVIEW

**Comment by dudycooly** [comment]
W.rt RFC, looks like  [Location](http://tools.ietf.org/html/rfc2616#section-14.30) is not applicable for [202-Accepted](http://tools.ietf.org/html/rfc2616#section-10.2.3). 

If so, we dont need to check  Location  for ACCEPTED ?!.

**Comment by dudycooly** [comment]
OrphanResource do have parent :).  We are forcing to become Orphan. Should we ?

**Comment by deontologician** [comment]
Lol! I didn't think of that. They have a parent but no url. Hmm. It's probably fine though

Sorry I haven't gotten a chance to look at this stuff. I am going to check it out tonight

File: restnavigator/halnav.py [hunk 1]
```diff
@@ -270,14 +283,81 @@ def create(self,
                 nav=self,
                 response=response,
             )
-        if response.status_code in (httplib.CREATED,
-                                    httplib.ACCEPTED,
+        return response
+
+    def create_navigator_or_orphan_resource_based_on_status_code(self, response):
```
Comment (deontologician) [inline restnavigator/halnav.py]: Maybe a shorted name like "navigator_or_orphan"?

File: restnavigator/halnav.py [hunk 2]
```diff
@@ -413,7 +495,8 @@ def fetch(self, *args, **kwargs):
             'Maybe you want this object\'s .parent attribute, '
             'or possibly one of the resources in .links')
 
-    __call__ = fetch
+    def __call__(self, *args, **kwargs):
```
Comment (deontologician) [inline restnavigator/halnav.py:497]: this is a great addition

File: restnavigator/halnav.py [hunk 3]
```diff
@@ -270,14 +283,81 @@ def create(self,
...
-                                    httplib.ACCEPTED,
+        return response
+
+    def create_navigator_or_orphan_resource_based_on_status_code(self, response):
+        if response.status_code in (httplib.CREATED, # Applicable for POST
                                     httplib.FOUND,
                                     httplib.SEE_OTHER,
-                                    ) and 'Location' in response.headers:
+                                    httplib.NO_CONTENT,
+                                    httplib.ACCEPTED # It doesnt belong here, under debate
```
Comment (deontologician) [inline restnavigator/halnav.py]: You're right, this shouldn't be in here. The RFC indicates that the body of the response should point to any status monitors etc, so Location on a 202 would be a custom hook.

**Comment by deontologician** [comment]
This looks good overall, just need to rename that super long method and remove the 202 checking as you indicated.

One other thing: could you change the target of this PR to the "next" branch? I want to merge this and the embedding stuff into a separate branch from master, and then pull it all in at once

**Comment by deontologician** [comment]
OK, I'm going to rebase this (squash intermediate commits) and accept it into next.

**Comment by dudycooly** [comment]
Fine. Not sure how to get rid of the intermediate commits without re-cloning

**Comment by deontologician** [comment]
Merged in with #17

**Pull request closed without merging**